common/src/main/java/com/wynntils/screens/maps/PoiManagementScreen.java

```diff
@@ -0,0 +1,243 @@
+                        this.width / 2f - 165,
+                        43,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        TextShadow.NORMAL);
+
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        I18n.get("screens.wynntils.poiManagementGui.name"),
+                        this.width / 2f - 130,
+                        43,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        TextShadow.NORMAL);
+
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        "X",
+                        this.width / 2f - 15,
+                        43,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        TextShadow.NORMAL);
+
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        "Y",
+                        this.width / 2f + 40,
+                        43,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        TextShadow.NORMAL);
+
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        "Z",
+                        this.width / 2f + 80,
+                        43,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        TextShadow.NORMAL);
+
+        RenderUtils.drawRect(poseStack, CommonColors.WHITE, this.width / 2 - 165, 52, 0, 355, 1);
+
+        poiManagerWidgets.forEach(widget -> widget.render(poseStack, mouseX, mouseY, partialTick));
+    }
+
+    @Override
+    public boolean mouseScrolled(double mouseX, double mouseY, double delta) {
+        if (delta < 0.0 && nextButton.active) {
+            nextPage();
+        } else if (delta > 0.0 && previousButton.active) {
+            previousPage();
+        }
+
+        return true;
+    }
+
+    public void populatePois() {
+        for (AbstractWidget widget : poiManagerWidgets) {
+            this.removeWidget(widget);
+        }
+
+        this.poiManagerWidgets.clear();
+
+        if (Math.min(pageHeight, waypoints.size() - pageHeight * page) == 0 && page != 0) {
+            previousPage();
+        }
+
+        for (int i = 0, lim = Math.min(pageHeight, waypoints.size() - pageHeight * page); i < lim; i++) {
+            CustomPoi poi = waypoints.get(page * pageHeight + i);
+
+            if (poi == null) {
```

Comment:
How can this happen?

Reply:
In theory it shouldn't, it's just a safety precaution in case it somehow does. Can remove it if you think it's pointless having it but I just thought better to be on the safe side

Reply:
Why did you resolve this, but not remove it?

Reply:
I have removed it?

https://github.com/Wynntils/Artemis/pull/1382/commits/ff6e943cb14eedd26a690b5d88abf33027b0105a#diff-908a218ec807dddcff8aababd8ececfbd7171a996e737521ef43a7fffb7b9ea6L182